src/economy/portdock.cc

```diff
@@ -327,134 +318,23 @@ void PortDock::shipping_item_returned(Game& game, ShippingItem& si) {
 	waiting_.push_back(si);
 }
 
-/**
- * A ship changed destination and is now coming to the dock. Increase counter for need_ship.
- */
-void PortDock::ship_coming(Ship& ship, bool affirmative) {
-	OPtr<Ship> s(&ship);
-	if (affirmative) {
-		if (ships_coming_.find(s) == ships_coming_.end()) {
-			ships_coming_.insert(s);
-		}
-	} else {
-		auto it = ships_coming_.find(s);
-		if (it != ships_coming_.end()) {
-			ships_coming_.erase(it);
+bool PortDock::load_one_item(Game& game, Ship& ship, const PortDock& dest) {
+	for (auto it = waiting_.begin(); it != waiting_.end(); ++it) {
+		if (it->get_destination(game) == &dest) {
+			ship.add_item(game, *it);
```

Comment:
TODO(k.halfmann): replace saccing with lookup if feasable

Reply:
Can you explain?

Reply:
Sorry for my typo, in geenral linear searches shoud be avoided and replaced by Map lookups or such.

Reply:
I'm reluctant to to even more refactoring than needed in feature freeze so I added a TODO that this should be done soonish after the release